projects/micro-dash/src/lib/object/get.ts

```diff
@@ -11,11 +11,11 @@ import { Nil } from "../interfaces";
  * - Micro-dash: 65 bytes
  */
 
-export function get<T, K1 extends keyof T, R>(
+export function get<T, K1 extends keyof T, R extends T[K1]>(
   object: T,
   path: [K1],
   defaultValue: R,
-): R;
+): R extends undefined ? T[K1] : Exclude<T[K1], undefined>;
```

Comment:
This looks good. It should be applied to all the overloads. And how 'bout calling it `D` for "default" instead of `R`?

Reply:
Sure. I was thinking `R` for `Result`, but it is just the default type.